src/main/java/com/webapp/TextBook/sharedFiles/ValidationBindingHelper.java

```diff
@@ -0,0 +1,181 @@
+package com.webapp.TextBook.sharedFiles;
+
+import com.sun.istack.Nullable;
+import com.webapp.TextBook.validation.Shared.SharedValidationState;
+import org.javatuples.Pair;
+import org.springframework.boot.configurationprocessor.json.JSONArray;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.boot.context.properties.bind.validation.BindValidationException;
+import org.springframework.ui.ModelMap;
+import org.springframework.validation.BindException;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ValidationBindingHelper {
+
+    private static final String failedValidationStatusMessage = "Input missing or invalid";
+    private static final String INTERNAL_VALIDATION_ERROR = "Internal Error: please contact IT.";
+
+    // creates local final strings for error and general errors key
+    private static final String ERRORS_KEY = "Errors";
+    private static final String GENERAL_ERRORS_KEY = "GeneralError";
+
+    public static Pair<Boolean,JSONObject> validationBindingHandler(@NotNull final BindingResult result, @NotNull final ModelMap map) throws JSONException {
+        JSONObject data = new JSONObject();
+        Boolean returnResult = null;
+        try {
+
+
+            if (result.hasErrors()) {
+                //binding errors present, load in Status/Error messages and return to login
+                if (result.getErrorCount() != 1) {
+                    throw new IllegalArgumentException("Internal Error: (WebControllerValidationBindingHelper" +
+                        ":validationBindingHandler) - " +
+                        "View model Retaining Multiple validation Bindings. " +
+                        "View model should only retain one class/type validation");
+                }
+                ObjectError loginUserInfoError = result.getAllErrors().get(0);
+
+
+                data.put("StatusMessage", failedValidationStatusMessage);
+
+                // invoke helper to set error state
+                ValidationBindingHelper.handleErrorMessageType(loginUserInfoError.getDefaultMessage(), data);
+
+                map.addAttribute("Data", data);
+                returnResult = false;
+            }
+            else{
+                returnResult = true;
+            }
+        }
+        catch(IllegalArgumentException i){
+            System.out.println();
+            returnResult = false;
+        }
+        catch(Exception e){
+            System.out.println("Internal Error: (WebControllerValidationBindingHelper" +
+                    ":validationBindingHandler) - " +
```

Comment:
if we're going to return in the finally block then a generic error must be set in the json object (data)
Can we do that here?